AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -79,11 +79,13 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
 
     public func query<M: Model>(_ modelType: M.Type,
                                 where predicate: QueryPredicate? = nil,
+                                sort sortInput: QuerySortInput? = nil,
```

Comment:
since we're introducing a new attribute, we should "append" it to the existing ones, meaning that it should come after `paginationInput`

Reply:
actually, I think in the review meeting I spoke the reason I put `sort` between `where` and `paginate`. Because in sql statement, this is the required order, first: `where`, second: `order by`, third: `offset` and `limit`. But your point also makes sense. What do you think on the two options? which is better?

Reply:
I recall this discussion during the review and agree with the current design: that `sort` should come before `paginate`, since that correctly communicates the order of operations internally: the results will be filtered by `predicate`, sorted by `sortInput`, and then paginated by `paginationInput`.

Because we are adding the parameter with a default value, it should still be a non-breaking change at call sites. Customers who want to adopt sorting will have to specify `sort` in the middle of their arguments, so that is a little bit odd, but I think that's a reasonable tradeoff for the overall clarity of the method signature.